Date_Operate.sh

```diff
@@ -0,0 +1,195 @@
+	echo -e "\n"	
+
+	echo "The youngest player: "
+	for p in "${youngest[@]}"; do echo "$p"; done
+	echo "They are $Age_max years old" 
+	echo -e "\n"	
+
+	echo "There are $less_than_20 players younger than 20, taking up $p1% of all players"
+	echo "There are $between_20_30 players aged between 20 and 30, taking up $p2% of all players"
+	echo "There are $greater_than_30 players older than 30, taking up $p3% of all players"
+}
+
+
+function PosInfo
+{
+	# 确定Position的列号
+	line=$(head -1 "$1")
+	IFS='	' read -r -a arr <<< "$line"
+	i=0
+	for w in "${arr[@]}"; do
+		if [[ "$w" == "Position" ]]; then 
+			break
+		fi
+		i=$((i+1))
+	done
+	
+	declare -A dic		## 定义字典，<球场位置，数量>
+	#dic=()
+	sum=0
+
+	# 从第2行开始读
+	ln=2
+	line="$(sed -n "$ln"p "$1")"
+	while [[ "$line" != "" ]]; do
+		sum=$((sum+1))
+		IFS='	' read -r -a arr <<< "$line"
+		pos=${arr["$i"]}
+		mark=0
+		for key in "${!dic[@]}"; do			# 判断字典中是否有键与$pos相同
+			if [[ "$pos" == "$key" ]]; then 
+				dic[$key]=$(( ${dic[$key]}+1 ))
+				mark=1
+				break
+			fi
+		done
+		if [[ "$mark" -ne 1 ]]; then
+			value=1 
+			dic[$pos]=$value
+		fi
+
+		ln=$((ln+1))
+		line="$(sed -n "$ln"p "$1")"
+	done
+	
+	printf "+--------------------------------+\n"
+	printf "| Position |  Amount  |  Rate(%%) |\n"
+	printf "+--------------------------------+\n"
+	for pos in "${!dic[@]}"; do 
+		r=$(echo "scale=4; ${dic[$pos]}/$sum*100" | bc)
+		printf "|%-10s|%-10s|%-10.2f|\n" "$pos" "${dic[$pos]}" "$r"
+		printf "+--------------------------------+\n" 
+	done
+	
+}
+
+
+function NameInfo
+{
+	shortest=100
+	sn=()
+	longest=0
+	ln=()
+	
+	# 确定Player的列号
+	line=$(head -1 "$1")
+	IFS='	' read -r -a arr <<< "$line"
+	i=0
+	for w in "${arr[@]}"; do
+		if [[ "$w" == "Player" ]]; then
+			break
+		fi
+		i=$((i+1))
+	done
+	
+	l=2
+	line=$(sed -n "$l"p "$1")
+	while [[ "$line" != "" ]]; do
+		IFS='	' read -r -a arr <<< "$line"
+		name=${arr[$i]}
+		
+		length=0
+		IFS=' ' read -r -a namarr <<< "$name"
+		for w in "${namarr[@]}"; do length=$((length+${#w}));done
+
+		# 出现更小的长度
+		if [[ $length -lt $shortest ]]; then shortest=$length sn=("$name");
+		# 名字长度相等
+		elif [[ $length -eq $shortest ]]; then
+			mark=0
+			for n in "${sn[@]}"; do 	# 去重
+				if [[ $n == "$name" ]]; then mark=1 break; fi
+			done
+			if [[ $mark -eq 0 ]]; then sn=(${sn[@]} "$name"); fi
+		fi
+		
+		# 出现更长的长度
+		if [[ $length -gt $longest ]]; then longest=$length ln=("$name");
+		# 名字长度相等
+		elif [[ $length -eq $longest ]]; then
+			mark=0
+			for n in "${ln[@]}"; do
+				if [[ $n == "$name" ]]; then mark=1 break; fi
+			done
+			if [[ $mark -eq 0 ]]; then ln=(${ln[@]} "$name"); fi
+		fi
+		
+		l=$((l+1))
+		line=$(sed -n "$l"p "$1")
+	done
+	
+	for p in "${ln[@]}"; do echo "$p"; done
+	printf "have the longest name ,length %s\n\n" "$longest"
+	for p in "${sn[@]}"; do echo "$p"; done
+	printf "have the shortest name ,length %s\n" "$shortest" 
+	
+}
+
+
+AgeInfo "worldcupplayerinfo.tsv"
+PosInfo "worldcupplayerinfo.tsv"
+NameInfo "worldcupplayerinfo.tsv"
```

Comment:
Line 65:
        echo -e "\n"        
                 ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 70:
        echo -e "\n"        
                 ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 119:
        printf "+--------------------------------+\n"
                                                  ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 120:
        printf "| Position |  Amount  |  Rate(%%) |\n"
                                                   ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 121:
        printf "+--------------------------------+\n"
                                                  ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 124:
                printf "|%-10s|%-10s|%-10.2f|\n" "$pos" "${dic[$pos]}" "$r"
                                             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 125:
                printf "+--------------------------------+\n" 
                                                          ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 167:
                        if [[ $mark -eq 0 ]]; then sn=(${sn[@]} "$name"); fi
                                                       ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
 
Line 178:
                        if [[ $mark -eq 0 ]]; then ln=(${ln[@]} "$name"); fi
                                                       ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
 
Line 186:
        printf "have the longest name ,length %s\n\n" "$longest"
                                                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                  ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
 
Line 188:
        printf "have the shortest name ,length %s\n" "$shortest" 
                                                 ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".